### DISCUSSION TITLE
nova2 - Spanish (special) characters are not being encoded correctly

### DISCUSSION THREAD

**Discussion by luisedelgado** [deepgram/community#694] [General help] [answered]
### Which Deepgram product feature are you using?

Deepgram API - STT Pre-recorded

### Details

Spanish characters such as [ü, é, á, í, ó, ú, ñ, Ñ] are not being encoded correctly, and therefore I'm seeing this type of response text "**Salt\u00f3** a la fama en su **pa\u00eds** natal en dos mil once"

How can I get Spanish characters to show up correctly? I'm enabling smartFormat and specifying language as 'es-419' in the request

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

dbaf6532-a61e-42d1-9b41-641bbeefc112

### If possible, please attach your code or paste it into the text box.

import os
from dotenv import load_dotenv

from deepgram import (
    DeepgramClient,
    PrerecordedOptions,
    FileSource,
)

load_dotenv()

# Path to the audio file
AUDIO_FILE = "data/Maluma-corto-con-puntuacion.wav"
API_KEY = os.getenv("DG_API_KEY")

def main():
    try:
        # STEP 1 Create a Deepgram client using the API key
        deepgram = DeepgramClient(API_KEY)

        with open(AUDIO_FILE, "rb") as file:
            buffer_data = file.read()

        payload: FileSource = {
            "buffer": buffer_data,
        }

        #STEP 2: Configure Deepgram options for audio analysis
        options = PrerecordedOptions(
            model="nova-2",
            smart_format=True,
            language="es-419",
        )

        # STEP 3: Call the transcribe_file method with the text payload and options
        response = deepgram.listen.prerecorded.v("1").transcribe_file(payload, options)

        # STEP 4: Print the response
        print(response.to_json(indent=4))

    except Exception as e:
        print(f"Exception: {e}")

if __name__ == "__main__":
    main()

### If possible, please attach an example audio file to reproduce the issue.

[Maluma-corto-con-puntuacion.wav.zip](https://github.com/deepgram/community/files/14948108/Maluma-corto-con-puntuacion.wav.zip)

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jpvajda** [comment]
@luisedelgado Thanks for letting us know about this. Let me share this with some of our engineering teams and see if this is known issue with our Nova-2 Spanish Model. 

I did try to test out your attached .wav file and I'm getting an error as it's exceeding our file limits for audio payload. 

```
{
  "err_code": "PAYLOAD_TOO_LARGE",
  "err_msg": "Payload size exceeds limit of 2 MB.",
}
```

It seems like that wav file is 3.5MB on disk.

I'm curious if you've tried to run this with `smart_format` set to false and with the language parameter of `es` yet?

**Comment by ricardolsmendes** [comment, marked as answer]
Hi @luisedelgado,
I managed to make it work for Portuguese characters by adding `ensure_ascii=False` to `response.to_json()`, as follows:
```python
    # STEP 4: Print the response
    print(response.to_json(ensure_ascii=False, indent=4))
```
Hope it helps!

**Comment by alexobviously** [comment]
@team-deepgram in case you didn't know, this is still broken on the pre-recorded API.

Also for anyone wondering, you can solve this in dart like: `utf8.decode(latin1.encode(weirdString))`
